plugins/plugin-java/che-plugin-java-ext-lang-server/src/main/java/org/eclipse/che/plugin/java/server/rest/RefactoringService.java

```diff
@@ -257,53 +258,128 @@ public RenameRefactoringSession createRenameRefactoring(CreateRenameRefactoring
         elementToRename = null;
     }
     if (elementToRename == null) {
-      throw new RefactoringException("Can't find java element to rename.");
+      throw new RefactoringException("Can't find java element to recommend.");
     }
 
+    JavaRenameRecommend.javaElement = elementToRename;
+
     return manager.createRenameRefactoring(
         elementToRename, cu, settings.getOffset(), settings.isRefactorLightweight());
   }
 
   /**
-   * Apply linked mode rename refactoring.
+   * Apply linked mode recommend refactoring.
    *
    * @param refactoringApply linked mode setting and refactoring session id
    * @return the result fo applied refactoring
    * @throws RefactoringException when there are no corresponding refactoring session
-   * @throws CoreException when impossible to apply rename refactoring
+   * @throws CoreException when impossible to apply recommend refactoring
    */
   @POST
-  @Path("rename/linked/apply")
+  @Path("recommend/linked/apply")
```

Comment:
can you avoid methods renaming? It will allow us to keep API backward compatible.